fe/fe-core/src/main/java/com/starrocks/meta/lock/LockManager.java

```diff
@@ -0,0 +1,521 @@
+// Copyright 2021-present StarRocks, Inc. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+package com.starrocks.meta.lock;
+
+import com.starrocks.common.Config;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+import java.util.ArrayList;
+import java.util.Comparator;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+public class LockManager {
+    private static final Logger LOG = LogManager.getLogger(LockManager.class);
+
+    private final int lockTablesSize;
+    private final Object[] lockTableMutexes;
+    private final Map<Long, Lock>[] lockTables;
+
+    public LockManager() {
+        lockTablesSize = Config.lock_table_num;
+        lockTableMutexes = new Object[lockTablesSize];
+        lockTables = new Map[lockTablesSize];
+        for (int i = 0; i < lockTablesSize; i++) {
+            lockTableMutexes[i] = new Object();
+            lockTables[i] = new HashMap<>();
+        }
+    }
+
+    /**
+     * Attempt to acquire a lock of 'lockType' on rid
+     *
+     * @param rid      The resource id to lock
+     * @param locker   The Locker to lock this on behalf of.
+     * @param lockType Then lock type requested
+     * @param timeout  milliseconds to time out after if lock couldn't be obtained. 0 means block indefinitely.
+     *
+     * @throws LockTimeoutException when the transaction time limit was exceeded.
+     * @throws DeadlockException when deadlock was detected
+     */
+
+    public void lock(long rid, Locker locker, LockType lockType, long timeout)
+            throws LockInterruptException, LockTimeoutException, DeadlockException {
+
+        final long startTime;
+
+        synchronized (locker) {
+            int lockTableIdx = getLockTableIndex(rid);
+            synchronized (lockTableMutexes[lockTableIdx]) {
+                Map<Long, Lock> lockTable = lockTables[lockTableIdx];
+                Lock lock = lockTable.get(rid);
+
+                if (lock == null) {
+                    lock = new LightWeightLock();
+                    lockTable.put(rid, lock);
+                } else if (lock instanceof LightWeightLock) {
+                    List<LockHolder> owners = new ArrayList<>(lock.getOwners());
```

Comment:
Since we need a list, how about changing the return type of `getOwners()` to `List<LockHolder>`?